src/app/project-admin/component/project-admin.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription, timer } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { NotificationService } from 'src/app/base/services/notification.service';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { RouteService } from 'src/app/base/services/route.service';
+import { UserManager } from 'src/app/util/user-manager';
+import { copyToClipboard } from 'src/app/util/helper-functions';
+import { ConfigManager } from 'src/app/base/services/config-service';
+
+@Component({
+  selector: 'project-admin-admin',
+  templateUrl: './project-admin.component.html',
+  styleUrls: ['./project-admin.component.scss']
+})
+
+export class ProjectAdminComponent implements OnInit {
+
+  projectName = new FormControl('');
+  project$: any;
+  projectQuery$: any;
+  project: any;
+  personalAccessTokens: any;
+  personalAccessTokensQuery$: any;
+  newToken: string;
+  tokenCopied: boolean = false;
+  tokenNameIsDuplicated: boolean;
+  subscriptions$: Subscription[] = [];
+
+  constructor(
+    private routeService: RouteService,
+    private activatedRoute: ActivatedRoute,
+    private projectApolloService: ProjectApolloService,
+    private router: Router
+  ) { }
+
+  ngOnInit(): void {
+    this.checkUserIsAuthorized();
+    this.routeService.updateActivatedRoute(this.activatedRoute);
+
+    const projectId = this.activatedRoute.parent.snapshot.paramMap.get('projectId');
+    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(projectId);
+    this.project$.subscribe((project) => this.project = project);
+
+    NotificationService.subscribeToNotification(this, {
+      projectId: projectId,
+      whitelist: ["pat"],
+      func: this.handleWebsocketNotification
+    });
+
+    let preparationTasks$ = [];
+    preparationTasks$.push(this.preparePersonalAccessTokensRequest(projectId));
```

Comment:
this doesn't make sense if the request isn't waited for 
we use this in conjunction with a fork join or a combine latest when we need to wait for some requests to be finished before proceeding.
both not needed here so just preparing should do it

Reply:
Refactored so that queries are called directly in onInit without preparation logic